### PR TITLE
Migrate GithubVulnerabilities2V3 to MSI

### PR DESCRIPTION
**Pull request by drewgillies**
Addresses: https://github.com/NuGet/Engineering/issues/5660

The job now uses MSIs to authenticate to blob storage.

### CODE REVIEW

**Review by advay26 — APPROVED** [review]

File: src/GitHubVulnerabilities2v3/Job.cs [hunk 1]
```diff
@@ -27,6 +28,7 @@ namespace GitHubVulnerabilities2v3
 {
     public class Job : JsonConfigurationJob, IDisposable
     {
+        private const string ManagedIdentityClientIdKey = "UserManagedIdentityClientId";
```
Comment (advay26) [inline src/GitHubVulnerabilities2v3/Job.cs:31]: The config changes use a different config key for MSIs `ManagedIdentityClientId` (without the 'User' prefix).  https://nuget.visualstudio.com/NuGetMicrosoft/_git/NuGetDeployment/pullrequest/2537

Comment (joelverhagen) [reply src/GitHubVulnerabilities2v3/Job.cs:31]: Just to add to this, there should be a shared const to refer to right? I imagine a lot of jobs need this config key.

Comment (advay26) [reply src/GitHubVulnerabilities2v3/Job.cs:31]: We have some constants here, including `ManagedIdentityClientId`: https://github.com/NuGet/NuGetGallery/blob/dev/src/NuGet.Services.Configuration/Constants.cs#L10

Even if we go with `UserManagedIdentityClientId` for this job, it probably makes sense to add that constant to the common location and then use that instead.

**Review by erdembayar — APPROVED** [review]

File: src/GitHubVulnerabilities2v3/Job.cs (hunk 1 above)
Comment (drewgillies) [reply src/GitHubVulnerabilities2v3/Job.cs]: Thanks for catching this. Updated.

**Review by erdembayar — APPROVED** [review]

**Review by joelverhagen — APPROVED** [review]

**Review by advay26 — APPROVED** [review]

**Pull request merged**